components/Resize-node.coffee

```diff
@@ -22,48 +23,107 @@ exports.getComponent = ->
     datatype: 'integer'
     description: 'New height'
     required: false
-  c.outPorts.add 'canvas',
-    datatype: 'string'
+
+  c.outPorts.add 'out',
+    datatype: 'all'
+    description: 'Resized buffer'
+  c.outPorts.add 'factor',
+    datatype: 'number'
+    description: 'Original over resized dimensions factor'
+    required: false
+  c.outPorts.add 'original',
+    datatype: 'object'
+    description: 'Original dimension'
+    required: false
+  c.outPorts.add 'resized',
+    datatype: 'object'
+    description: 'Resized dimension'
+    required: false
+  c.outPorts.add 'metadata',
+    datatype: 'object'
+    description: 'Extracted metadata while resizing'
+    required: false
   c.outPorts.add 'error',
     datatype: 'object'
     required: false
 
   noflo.helpers.WirePattern c,
-    in: ['path']
+    in: ['in']
     params: ['width', 'height']
```

Comment:
Not a change here, but note that this means the component can never resize images to different sizes concurrently (cause params are per-component, shared between process() calls).